pkg/resource/rule/hooks.go

```diff
@@ -133,4 +133,21 @@ func customPreCompare(
 	if !equalTargets(desired.ko.Spec.Targets, latest.ko.Spec.Targets) {
 		delta.Add("Spec.Targets", desired.ko.Spec.Targets, latest.ko.Spec.Targets)
 	}
+
+	// ideally EqualStrings should do but we're missing one case there (see
+	// EqualStrings function comment)
+	desiredExpression := desired.ko.Spec.ScheduleExpression
+	latestExpression := latest.ko.Spec.ScheduleExpression
+	if !(pkgtags.EqualZeroString(desiredExpression) && pkgtags.EqualZeroString(latestExpression)) {
+		delta.Add("Spec.ScheduleExpression", desiredExpression, latestExpression)
+	}
```

Comment:
Are we missing the case where `*desiredExpression != *latestExpression`?

Reply:
We do 😶‍🌫️ (fixed w/ unit tests)